modules/identity/src/Volo.Abp.PermissionManagement.Domain.Identity/Volo/Abp/PermissionManagement/Identity/RoleUpdateEventHandler.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Threading.Tasks;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.Domain.Entities.Events;
+using Volo.Abp.EventBus;
+using Volo.Abp.Identity;
+
+namespace Volo.Abp.PermissionManagement.Identity
+{
+    public class RoleUpdateEventHandler :
+        ILocalEventHandler<IdentityRoleNameChangedEvent>,
+        ITransientDependency
+    {
+        protected IIdentityRoleRepository RoleRepository { get; }
+        protected IPermissionManager PermissionManager { get; }
+        protected IPermissionGrantRepository PermissionGrantRepository { get; }
+
+        public RoleUpdateEventHandler(
+            IIdentityRoleRepository roleRepository,
+            IPermissionManager permissionManager,
+            IPermissionGrantRepository permissionGrantRepository)
+        {
+            RoleRepository = roleRepository;
+            PermissionManager = permissionManager;
+            PermissionGrantRepository = permissionGrantRepository;
+        }
+
+        public async Task HandleEventAsync(IdentityRoleNameChangedEvent eventData)
+        {
+            var role = await RoleRepository.FindAsync(eventData.IdentityRole.Id, false).ConfigureAwait(false);
+            if (role == null)
+            {
+                return;
+            }
+
+            var permissionGrantsInRole = await PermissionGrantRepository.GetListAsync("R", eventData.OldName).ConfigureAwait(false);
+            foreach (var permissionGrant in permissionGrantsInRole)
+            {
+                await PermissionManager.UpdateProviderKeyAsync(permissionGrant, eventData.IdentityRole.Name).ConfigureAwait(false);
```

Comment:
I'm not sure that the `IPermissionManager` should add the` UpdateProviderKeyAsync` method.

Maybe it can use `IPermissionGrantRepository` directly. so some changes below can be undone.

Reply:
ok. I can move in the repo.

Reply:
It seems that we have ignored the above discussion. : )